### PR TITLE
First version of parse-print analysis script.

### LINKED ISSUE #787: Create script for testing the parse-print cycle.
Given a list of git-based projects, for each of them do the following:
- Download a copy of the project.
- For each source file:
  - Run through a Prettifier pass with our config.
  - Parse, then print the model.
  - Create a diff against the prettified version.

### PR DESCRIPTION
**Pull request by seanparsons**
Fixes #787
**Problem:**
It's hard to gauge how well the parser printer combination works currently without manually testing projects against it.

**Fix:**
Created a script that can be run with `npm run parse-print-analysis`, which will produce a bunch of diffs for files in those projects so that issues can be identified and then retested subsequently.

**Commit Details:**
- Fixes #787.
- Created `parse-print-analysis.ts` file which processes files in a
  collection of projects to see the result of parsing and then
  printing a JavaScript file.
- Fixed one small issue with the parser not allowing fragments as a
  top level element of a component.
- Added `parse-print-analysis` npm script.

### CODE REVIEW

File: editor/package.json [hunk 1]
```diff
@@ -30,7 +30,8 @@
     "install-utopia-api": "cd ../utopia-api && npm install && npm run build && npm run dts-bundle",
     "preinstall": "npm run install-utopia-api",
     "postinstall": "patch-package",
-    "eslint-check": "eslint --quiet --config .eslintrc.js --ext .ts --ext .tsx src ../utopia-api/src ../website/src"
+    "eslint-check": "eslint --quiet --config .eslintrc.js --ext .ts --ext .tsx src ../utopia-api/src ../website/src",
+    "parse-print-analysis": "npx ts-node --transpile-only ./src/scripts/parse-print-analysis.ts"
```
Comment (balazsbajorics) [inline editor/package.json:34]: yay for ts-node!

**Comment by maltenuhn** [comment]
Noice!

**Review by enidemi — APPROVED** [review]

File: editor/src/scripts/parse-print-analysis.ts [hunk 2]
```diff
@@ -0,0 +1,157 @@
+/* eslint no-console: "off" */
+import * as Path from 'path'
+import * as FS from 'fs'
```
Comment (balazsbajorics) [inline editor/src/scripts/parse-print-analysis.ts:3]: path and fs are usually imported as lower case in node scripts, when I read `FS.readFileSync` in the code I thought you are using a third party library that wraps `fs`

File: editor/.dependency-cruiser.js [hunk 3]
```diff
@@ -159,6 +159,19 @@ module.exports = {
         path: '\\.(test-utils)\\.(js|mjs|cjs|ts|tsx|ls|coffee|litcoffee|coffee\\.md)$',
       },
     },
+    {
+      name: 'not-to-scripts',
```
Comment (Rheeseyb) [inline editor/.dependency-cruiser.js:163]: Nice!

File: editor/.dependency-cruiser.js [hunk 4]
```diff
@@ -171,7 +184,7 @@ module.exports = {
       from: {
         path: '^(src)',
         pathNot:
-          '\\.(spec|test|spec.browser|test-utils)\\.(js|mjs|cjs|ts|tsx|ls|coffee|litcoffee|coffee\\.md)$',
+          '\\.(spec|test|spec.browser|test-utils)\\.(js|mjs|cjs|ts|tsx|ls|coffee|litcoffee|coffee\\.md)$|(src/scripts/.*$)',
```
Comment (Rheeseyb) [inline editor/.dependency-cruiser.js:187]: My eyes!

File: editor/src/core/workers/parser-printer/parser-printer.ts [hunk 5]
```diff
@@ -457,7 +457,7 @@ function printUtopiaJSXComponent(
   detailOfExports: ExportsDetail,
 ): TS.Node {
   const asJSX = jsxElementToExpression(element.rootElement, imports, printOptions.stripUIDs)
-  if (TS.isJsxElement(asJSX) || TS.isJsxSelfClosingElement(asJSX)) {
+  if (TS.isJsxElement(asJSX) || TS.isJsxSelfClosingElement(asJSX) || TS.isJsxFragment(asJSX)) {
```
Comment (Rheeseyb) [inline editor/src/core/workers/parser-printer/parser-printer.ts:460]: Is this a bug fix?

File: editor/src/scripts/parse-print-analysis.ts (hunk 2 above)
Comment (Rheeseyb) [reply editor/src/scripts/parse-print-analysis.ts:3]: +1 to this, as silly as the convention seems

**Review by Rheeseyb — APPROVED** [review]

File: editor/src/scripts/parse-print-analysis.ts (hunk 2 above)
Comment (seanparsons) [reply editor/src/scripts/parse-print-analysis.ts:3]: I'll be doing this in a subsequent PR as I've already made a bunch of changes to this there.

File: editor/src/core/workers/parser-printer/parser-printer.ts (hunk 5 above)
Comment (seanparsons) [reply editor/src/core/workers/parser-printer/parser-printer.ts:460]: It is, mentioned this in the description above.

File: editor/.dependency-cruiser.js (hunk 4 above)
Comment (seanparsons) [reply editor/.dependency-cruiser.js:187]: The horrifying burns inflicted by regular expressions.

**Pull request merged**